service/web/subscriberserver.go

```diff
@@ -47,6 +47,24 @@ func (s subscriberserver) CreateSubscription(ctx context.Context, req *metrov1.S
 	return req, nil
 }
 
+// UpdateSubscription updates a given subscription
+func (s subscriberserver) UpdateSubscription(ctx context.Context, req *metrov1.UpdateSubscriptionRequest) (*metrov1.Subscription, error) {
+	logger.Ctx(ctx).Infow("subscriberserver: received request to update subscription", "name", req.Subscription.Name, "topic", req.Subscription.Topic)
+	span, ctx := opentracing.StartSpanFromContext(ctx, "SubscriberServer.UpdateSubscription")
+	defer span.Finish()
+
+	m, paths, err := subscription.GetValidatedModelAndPathsForUpdate(ctx, req)
+	if err != nil {
+		return nil, merror.ToGRPCError(err)
+	}
+
+	m, err = s.subscriptionCore.UpdateSubscription(ctx, m, paths)
+	if err != nil {
+		return nil, merror.ToGRPCError(err)
+	}
+	return subscription.ModelToSubscriptionProtoV1(m), nil
```

Comment:
can we just return req.subcription as response ?

Reply:
> @razorao Did a quick setup of this branch on my local. This was the flow I had in mind using the already available `proto-helper-functions` and `fmutils` package.
> 
> ```
> // step 1
> updateSubscriptionRequest.Fieldmask.Normalize()
> 
> // step 2
> if !updateSubscriptionRequest.Fieldmask.IsValid(updateSubscriptionRequest.Subscription) {
> 		// Return an error.
> }
> 
> // step 3 Redact the request according to the provided field mask.
> fmutils.Filter(updateSubscriptionRequest.GetSubscription(), updateSubscriptionRequest.Fieldmask.GetPaths())
> 
> // step 4 Now that the request is vetted we can merge it with the Subscription entity.
> proto.Merge(existingSubscriptionAsProto, updateProfileRequest.GetSubscription())
> NOTE: savedDBSubscription to Proto and vice-versa is the additional needed mapping step.
> ```
> 
> Do have a look at these and see if these reduce custom code addition and increase re-usability across models for the future.

@shivasishdas Tried your suggestion
The problem is when we are tying to patch with an emtpy value, Eg: Patch an int with 0, or patch a pointer with nil,
The Merge method does not  apply the patch.

Reply:
@razorao Few questions:

- Whats happens on patch `int with 0` or `a string with ""`, do they simply get ignored? Asking as these seem to be default value cases for primitives.
- Does update work fine for non-default primitive values?

Reply:
patching int with 0 does not work, string I did not check, But I assume it's the same.
Update works fine for non empty values.. patching int from 100 -> 50 works, 100 -> 0 breaks
push_config: {"endpoint": "endpoint1"} -> push_config: {"endpoint": "endpoint2"} works
push_config: {"endpoint": "endpoint1"} -> nil, nope!

Reply:
From what I've seen, google pub sub returns the actual data rather than what the client provides.
I tried out a scenario, where I do not provide the `ackDeadlineSeconds` attribute in the request (because I don't intend to update that). In the response, I receive the `ackDeadlineSeconds` as 100 (which is it's real value)